src/main/scala/com/cognite/sdk/scala/common/ReadableResource.scala

```diff
@@ -7,69 +7,99 @@ import fs2._
+  def readFromCursorWithLimit(
+      cursor: String,
+      limit: Long,
+      partition: Partition = Partition()
+  ): F[ItemsWithCursor[R]] =
+    readWithCursor(Some(cursor), Some(limit), partition)
 
-  def read(): F[ItemsWithCursor[R]] = readWithCursor(None, None)
+  def read(partition: Partition = Partition()): F[ItemsWithCursor[R]] =
+    readWithCursor(None, None, Partition())
 
-  def readWithLimit(limit: Long): F[ItemsWithCursor[R]] =
-    readWithCursor(None, Some(limit))
+  def readWithLimit(limit: Long, partition: Partition = Partition()): F[ItemsWithCursor[R]] =
+    readWithCursor(None, Some(limit), partition)
 
   private def listWithNextCursor(
       cursor: Option[String],
-      limit: Option[Long]
+      limit: Option[Long],
+      partition: Partition
   ): Stream[F, R] =
-    Readable.pullFromCursorWithLimit(cursor, limit, readWithCursor).stream
+    Readable.pullFromCursorWithLimit(cursor, limit, partition, readWithCursor).stream
 
-  def listFromCursor(cursor: String): Stream[F, R] =
-    listWithNextCursor(Some(cursor), None)
+  def listFromCursor(cursor: String, partition: Partition = Partition()): Stream[F, R] =
+    listWithNextCursor(Some(cursor), None, partition)
 
-  def listWithLimit(limit: Long): Stream[F, R] =
-    listWithNextCursor(None, Some(limit))
+  def listWithLimit(limit: Long, partition: Partition = Partition()): Stream[F, R] =
+    listWithNextCursor(None, Some(limit), partition)
 
-  def listFromCursorWithLimit(cursor: String, limit: Long): Stream[F, R] =
-    listWithNextCursor(Some(cursor), Some(limit))
+  def listFromCursorWithLimit(
+      cursor: String,
+      limit: Long,
+      partition: Partition = Partition()
+  ): Stream[F, R] =
+    listWithNextCursor(Some(cursor), Some(limit), partition)
+
+  def list(partition: Partition = Partition()): Stream[F, R] =
+    listWithNextCursor(None, None, partition)
 
-  def list(): Stream[F, R] = listWithNextCursor(None, None)
+  def listPartitions(numPartitions: Int): Seq[Stream[F, R]] =
+    Seq(listWithNextCursor(None, None, Partition()))
 }
 
 object Readable {
   @SuppressWarnings(Array("org.wartremover.warts.Recursion"))
   private[common] def pullFromCursorWithLimit[F[_], R](
       cursor: Option[String],
       limit: Option[Long],
-      get: (Option[String], Option[Long]) => F[ItemsWithCursor[R]]
+      partition: Partition,
+      get: (Option[String], Option[Long], Partition) => F[ItemsWithCursor[R]]
   ): Pull[F, R, Unit] =
     if (limit.exists(_ <= 0)) {
       Pull.done
     } else {
-      Pull.eval(get(cursor, limit)).flatMap { items =>
+      Pull.eval(get(cursor, limit, partition)).flatMap { items =>
         Pull.output(Chunk.seq(items.items)) >>
           items.nextCursor
-            .map(s => pullFromCursorWithLimit(Some(s), limit.map(_ - items.items.size), get))
+            .map { s =>
+              pullFromCursorWithLimit(Some(s), limit.map(_ - items.items.size), partition, get)
```

Comment:
Could this end up reading more than limit?

Reply:
It's the classic "limit is per partition" problem, I'll see if I can address that somehow. I think the interface currently doesn't allow specifying limit when using partitions, which it absolutely should.